spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/BasicAuthSecurityConfiguration.java

```diff
@@ -157,6 +161,9 @@ protected void configure(HttpSecurity http) throws Exception {
 		http.addFilterBefore(sessionRepositoryFilter,
 				ChannelProcessingFilter.class).csrf().disable();
 		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED);
+
+		securityStateBean.setAuthenticationEnabled(true);
+		securityStateBean.setAuthorizationEnabled(true);
```

Comment:
This is set to `false` in `OAuthSecurityConfiguration` and looks like assumption is that `OnSecurityEnabledAndOAuth2Enabled` protects that only one config is enabled. I can't wrap my head around how `OnSecurityEnabledAndOAuth2Enabled` now works that `security.basic.enabled` has been removed from it. Am I right that `BasicAuthSecurityConfiguration` is called anyway and if `security.oauth2.client.client-id` is set, also `OAuthSecurityConfiguration` is called and `SecurityStateBean` parameters are set twice. In that case we're rely on order how config classes are called?